File: javascript/CartItem.js
// javascript/CartItem.js
export class CartItem {
    constructor(product, quantity) {
        this.product = product;
        this.quantity = quantity;
    }

    getProduct() {
        return this.product;
    }

    getQuantity() {
        return this.quantity;
    }

    getTotalPrice() {
        return this.product.getPrice() * this.quantity;
    }

    setQuantity(quantity) {
        this.quantity = quantity;
    }
}

File: cpp/CartItem.cpp
#include "CartItem.h"

CartItem::CartItem(Product* product, int quantity)
    : product(product), quantity(quantity) {}

Product* CartItem::getProduct() const {
    return product;
}

int CartItem::getQuantity() const {
    return quantity;
}

double CartItem::getTotalPrice() const {
    return product->getPrice() * quantity;
}

void CartItem::setQuantity(int quantity) {
    this->quantity = quantity;
}
